src/renderer/src/lib/Music.ts

```diff
@@ -156,6 +130,105 @@ export async function next() {
   }
 
   setSong(current.song);
+  await setMediaSessionMetadata();
+  setMediaSessionPosition();
+}
+
+export async function setMediaSession(song: Song) {
+  const bgPath = await window.api.request("resource::getMediaSessionImage", song.bg!);
+  if (bgPath.isNone) {
+    return;
+  }
+
+  if ("mediaSession" in navigator) {
+    await setMediaSessionMetadata();
+    setMediaSessionPosition();
+
+    const actionHandlers = [
```

Comment:
I am not really fond of this section of code. I feel like repeating:
```javascript
try {
  navigator.mediaSession.setActionHandler("action", handler);
} catch (err) {
  console.log(`The media session action "action" is not supported yet.`);
}
```
would be better because this multidimensional array takes longer to understand imo

Reply:
fwiw you could also do
```ts
const actionHandlers = {
  "play": togglePlay,
  "pause": pause,
  "previoustrack": previous,
  "nexttrack": next,
}

for (const [action, handler] of Object.entries(actionHandlers)) {
```
but it's really not that big of a difference from just repeating try catch here yea

Reply:
siro, could you elaborate more on what gripes you have with the code there, i didnt quite understand

Reply:
brw's suggestion is my preference. But I'm not totally against Plextora's implementation.

Reply:
Yeah, I agree that his suggestion is good. I just implemented that in https://github.com/Team-BTMC/osu-radio/pull/16/commits/0b9f3c21a5e3d9e51bf8178482ed980b5984f7da